### PR TITLE
#415 linter preventing prs

### PR DESCRIPTION
**Pull request by grimley517**
A collection of small changes to comply with lining rules.

Also added folders - dist and build to the .gitignore.  These are generated in the CI pipeline. adding to gitignore prevents a user from accidentally committing these whilst working through pipeline.

### CODE REVIEW

**Comment by grimley517** [comment]
addresses issue #415

**Comment by grimley517** [comment]
In relation to patterns/behavioral/strategy.py

On adding the additional spaces required for flake linter to work, the line became too long.

```
$ flake8 . --count --show-source --statistics
./patterns/behavioral/strategy.py:59:121: E501 line too long (121 > 120 characters)
        return f"<Order price: {self.price} with discount strategy: {getattr(self.discount_strategy, '__name__', None)}>"
                                                                                                                        ^
1     E501 line too long (121 > 120 characters)
```

Hence changed to multiline comment

**Comment by grimley517** [comment]
Multiline string fails as generates blank lines.  

Extract lengthy call to variable - works better anyway as breaks up cognitive load of line

**Comment by grimley517** [comment]
The last commit isn't really within the scope of the issue. 

By extracting the linting and testing to an external script it is easier for developers to run this prior to deployment and see issues.

The linting and testing is still incomplete.  
4 tests still fail.  But these are beyond simple linting and should be discussed separately.

```
FAILED patterns/creational/abstract_factory.py::patterns.creational.abstract_factory.main
FAILED patterns/dependency_injection.py::patterns.dependency_injection.main
FAILED patterns/fundamental/delegation_pattern.py::patterns.fundamental.delegation_pattern.Delegator
FAILED patterns/other/blackboard.py::patterns.other.blackboard.main
============================================================================================== 4 failed, 98 passed in 0.29s ==============================================================================================
```

**Comment by grimley517** [comment]
willing to convert lint script to powershell or remove if requested.

**Comment by grimley517** [comment]
Raised #418 for first failing test.

File: lint.sh
Comment (grimley517) [inline lint.sh:1]: Not massively convinced this is the right name.

**Comment by grimley517** [comment]
```
FAILED patterns/dependency_injection.py::patterns.dependency_injection.main
```
This is due to a method being called which isn't there. Much more straightforward than the factory pattern error.

File: lint.sh
Comment (faif) [reply lint.sh:1]: Agreed, any proposals?

**Pull request merged**